src/recipe_dictionary.cpp

```diff
@@ -32,10 +32,121 @@ const recipe &recipe_dictionary::get_uncraft( const itype_id &id )
+        [r]( const std::string & tool ) {
+            const auto c = r->requirements().get_tools();
+            for( const auto &t : c ) {
+                for( auto z : t )
+                    if( lcmatch( z.to_string() , tool ) ) {
+                        return true;
+                    }
+            }
+            return false;
+        } );
+    };
+    auto result_quality_match = [result_qualities]( const recipe * r ) {
+        return std::any_of( result_qualities.begin(),
+                            result_qualities.end(),
+        [r]( const std::string & result_quality ) {
+            auto a = item( r->result );
+            return std::any_of( a.type->qualities.begin(),
+                                a.type->qualities.end(),
+            [result_quality]( std::pair<quality_id, int> quality ) {
+                return lcmatch( quality.first->name, result_quality );
+            } );
+        } );
+    };
+    auto requirement_quality_match = [requirement_qualities]( const recipe * r ) {
+        return std::any_of( requirement_qualities.begin(),
+                            requirement_qualities.end(),
+        [r]( const std::string & requirement_quality ) {
+            const auto &req_sets = r->requirements().get_qualities();
+            for( auto req_set : req_sets )
+                if( std::any_of( req_set.begin(),
+                                 req_set.end(),
+                [requirement_quality]( const quality_requirement & qr ) {
+                return lcmatch( qr.type->name, requirement_quality );
+                } ) ) {
+                return true;
+            }
+            return false;
+        } );
+    };
+    auto component_match = [components]( const recipe * r ) {
+        return std::any_of( components.begin(),
+                            components.end(),
+        [r]( const std::string & component ) {
+            const auto &required_comp = r->requirements().get_components();
+            return std::any_of( required_comp.begin(),
+                                required_comp.end(),
+            [component]( const std::vector<item_comp> &items ) {
+                return std::any_of( items.begin(), items.end(),
+                [component]( const item_comp & comp ) {
+                    return lcmatch( item::nname( comp.type ), component );
+                } );
+            } );
+        } );
+    };
+    auto skill_match = [skills]( const recipe * r ) {
+        return std::any_of( skills.begin() , skills.end(),
+        [r]( const std::string & skill ) {
+            return lcmatch( r->required_skills_string(), skill );
+        } );
+    };
     std::copy_if( recipes.begin(), recipes.end(), std::back_inserter( res ),
-    [ &txt ]( const recipe * r ) {
-        return lcmatch( item::nname( r->result ), txt );
+    [&]( const recipe * r ) {//sadly it seems easier to copy the scope here for the moment
+        return name_match( r ) || tools_match( r ) ||
```

Comment:
It would be clearer if you moved those earlier named lambdas here as they are used only once and it would help avoid some of your nesting problems